src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2740,78 +2771,90 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
 #End Region
 
-        Private Class SymbolSearcher
-
-            Private ReadOnly _cache As Dictionary(Of Declaration, NamespaceOrTypeSymbol)
+        Private MustInherit Class AbstractSymbolSearcher
```

Comment:
Note: this logic is nearly (or exactly) identical to the C# version.  We could unify this with one type.  Let me know if you want me to do that.

Reply:
Factoring the type sounds good to me.

---
In reply to: [183260588](https://github.com/dotnet/roslyn/pull/26325#discussion_r183260588) [](ancestors = 183260588)